mxcubeweb/core/components/harvester.py

```diff
@@ -0,0 +1,505 @@
+                return False
+        else:
+            logging.getLogger("user_level_log").error("Pin Calibration Failed")
+            logging.getLogger("user_level_log").error(
+                "Sample Changer could not mount Pin"
+            )
+            return False
+
+    def cancel_calibration(self) -> None:
+        harvester_device = HWR.beamline.harvester
+        harvester_device.set_calibrate_state(False)
+        logging.getLogger("user_level_log").warning("Pin Calibration Canceled")
+
+    def validate_calibration(self) -> bool:
+        """
+        finish Calibration Procedure step 2
+        after user ran a 3 click centring
+        Return (bool): whether the step 2 of calibration procedure
+        goes to end (True)
+        or had and exception (False)
+        """
+        try:
+            harvester_device = HWR.beamline.harvester
+            md = HWR.beamline.diffractometer
+
+            motor_pos_dict = {
+                "focus": md.focusMotor.get_value(),
+                "phiy": md.phiyMotor.get_value(),
+                "phiz": md.phizMotor.get_value(),
+                "centring_focus": md.centringFocus.get_value(),
+                "centring_vertical": md.centringVertical.get_value(),
+            }
+
+            saved_position = md.saved_motor_position
+            # find offset position based on old and new motor position
+            new_motor_offset = {
+                "focus": motor_pos_dict["focus"] - saved_position["focus"],
+                "phiy": motor_pos_dict["phiy"] - saved_position["phiy"],
+                "phiz": motor_pos_dict["phiz"] - saved_position["phiz"],
+                "centring_focus": (
+                    motor_pos_dict["centring_focus"] - saved_position["centring_focus"]
+                ),
+                "centring_vertical": (
+                    motor_pos_dict["centring_vertical"]
+                    - saved_position["centring_vertical"]
+                ),
+            }
+
+            calibrated_motor_offset = {
+                "focus": new_motor_offset["focus"] + new_motor_offset["centring_focus"],
+                "phiy": new_motor_offset["phiy"],
+                "phiz": (
+                    new_motor_offset["phiz"] + new_motor_offset["centring_vertical"]
+                ),
+            }
+
+            print(calibrated_motor_offset)
+            # we store the motor offset in the Harvester, to be used for sample centering
+            harvester_device.store_calibrated_pin(
+                calibrated_motor_offset["focus"],
+                calibrated_motor_offset["phiy"],
+                calibrated_motor_offset["phiz"],
+            )
+
+            harvester_device.set_calibrate_state(False)
+        except Exception as ex:
+            logging.getLogger("user_level_log").exception(
+                f"Pin Calibration / validation Failed {str(ex)}"
+            )
+            return False
```

Comment:
Another catch-all-exceptions? Why? At least there is logging, though...


```suggestion
        except Exception:
            logging.getLogger("user_level_log").exception(
                "Pin Calibration / validation Failed"
            )
            return False
```

No, but seriously why not list the actual exceptions we want to catch here?

```
except (ExceptionFoo, ExceptionBar):
```

and of course, there should be much less coed in the `try` block. 

I know those things are not easy, but seeing code like this scares me. There is a very high risk of us staying naively unaware of a whole bunch of issues that are actually very easy to fix if we keep having this poor handling of exceptions.